tests/test_matrix_utils_connect_e2ee.py

```
@@ -0,0 +1,690 @@
+            mock_client_instance.get_displayname = AsyncMock(
+                return_value=MagicMock(displayname="TestBot")
+            )
+            mock_async_client.return_value = mock_client_instance
+
+            # Create mock modules where nio.crypto lacks OlmDevice
+            mock_olm = SimpleNamespace()
+            mock_nio_crypto = SimpleNamespace()
+            # Simulate missing OlmDevice attribute to exercise hasattr failure
+            mock_nio_store = SimpleNamespace(SqliteStore=MagicMock())
+
+            def import_side_effect(name):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `import_side_effect`

(ANN202)